eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1701,16 +1701,18 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     // a channel_reestablish when reconnecting a channel that recently got confirmed, and instead send a funding_locked
     // first and then go silent. This is due to a race condition on their side, so we trigger a reconnection, hoping that
     // we will eventually receive their channel_reestablish.
-    case Event(_: FundingLocked, _) =>
+    case Event(_: FundingLocked, d) =>
       log.warning("received funding_locked before channel_reestablish (known lnd bug): disconnecting...")
+      send(Warning(d.channelId, "spec violation: you sent funding_locked before channel_reestablish"))
       peer ! Peer.Disconnect(remoteNodeId)
       stay
 
     // This handler is a workaround for an issue in lnd similar to the one above: they sometimes send announcement_signatures
     // before channel_reestablish, which is a minor spec violation. It doesn't halt the channel, we can simply postpone
     // that message.
-    case Event(remoteAnnSigs: AnnouncementSignatures, _) =>
+    case Event(remoteAnnSigs: AnnouncementSignatures, d) =>
       log.warning("received announcement_signatures before channel_reestablish (known lnd bug): delaying...")
+      send(Warning(d.channelId, "spec violation: you sent announcement_signatures before channel_reestablish"))
       context.system.scheduler.scheduleOnce(5 seconds, self, remoteAnnSigs)
       stay
```

Comment:
```suggestion
      val warning = Warning(d.channelId, "spec violation: you sent announcement_signatures before channel_reestablish")
      context.system.scheduler.scheduleOnce(5 seconds, self, remoteAnnSigs)
      stay sending warningy
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1863/commits/bf10f462f50cd45c865e59370b572a42f736cd39